File: routes/user/address.js
var cates = require("./models/cates.js");
var res = require('./lib/resp.js');
var session = require('./lib/session.js');
var users = require("./models/users.js");


var user = session.getUser();
if (!user) {
  return res.redirect('/routes/user/login.js');
}

if (request.getMethod() === 'GET') {
  var allData = cates.find();
  var data = {
    tree: cates.getTree(allData),
    user: user,
    cart: session.getCart(),
  };
  return res.render('./views/user/address.ejs', data);
}

if (request.getMethod() === 'POST') {
  var address = request.formValue('address');
  var name = request.formValue('name');
  var phone = request.formValue('phone');
  var addr = (address + '  ' + name + '  ' + phone).trim();

  var r = users.updateByOid(user.oid, { address: addr });
  console.log('%j', r);
  user.address = addr;
  session.set('user', JSON.stringify(user));
  res.redirect('./address.js');
}

File: routes/user/address.go
package user

import (
	"fmt"
	"strings"

	"github.com/zengming00/go-testShop/framework"
	"github.com/zengming00/go-testShop/lib"
)

func Address(ctx *framework.HandlerContext) {
	var user = ctx.GetUser()
	if user == nil {
		ctx.Redirect("/routes/user/login.go")
		return
	}
	if ctx.R.Method == "GET" {
		var allData = ctx.Cates.Find()
		var data = map[string]interface{}{
			"tree": ctx.Cates.GetTree(allData),
			"user": user,
			"cart": ctx.GetCart(),
		}
		ctx.Render("./views/user/address.html", data, lib.FuncMap)
		return
	}
	if ctx.R.Method == "POST" {
		var address = ctx.R.FormValue("address")
		var name = ctx.R.FormValue("name")
		var phone = ctx.R.FormValue("phone")
		var addr = strings.TrimSpace(address + "  " + name + "  " + phone)

		var r = ctx.Users.UpdateByOid(user.Oid, map[string]interface{}{"address": addr})
		fmt.Printf("%#v\n", r)
		user.Address = addr
		ctx.SetUser(user)
		ctx.Redirect("./address.go")
	}
}
